ably/realtime_channel.go

```diff
@@ -896,6 +918,86 @@ func (c *RealtimeChannel) notify(msg *protocolMessage) {
 	}
 }
 
+// processMessageWithDelta processes a message with delta decoding support (RTL19, RTL20).
+func (c *RealtimeChannel) processMessageWithDelta(msg *Message) error {
+	c.mtx.Lock()
+	defer c.mtx.Unlock()
+
+	// Update decoding context with current state (RTL19, RTL20)
+	c.decodingContext.BasePayload = c.basePayload
+	c.decodingContext.LastMessageID = c.lastMessageID
+
+	// Decode message with delta support
+	var cipher channelCipher
+	if c.options != nil {
+		cipher, _ = (*protoChannelOptions)(c.options).GetCipher()
+	}
+	decodedMsg, newBasePayload, err := msg.withDecodedDataAndContext(cipher, c.decodingContext)
+	if err != nil {
+		// Check for specific delta-related errors
+		if containsErrorCode(err, 40018) || containsErrorCode(err, 40019) {
```

Comment:
Not a go expert, but searching for error code inside string doesn't look right

Reply:
Yes I agree, we should either type assert the error to an ErrorInfo, or return specific error values and use [`errors.Is`](https://pkg.go.dev/errors#Is).